### PR TITLE
feat: Cluster node details action buttons

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

These changes will add an action buttons to the cluster node details which will contain different actions like,
- Cordon node
- Drain node
- Edit taints
- Edit YAML
- Delete node

Fixes - [AB#1613](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1613)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by visiting `Clusters > Select cluster > Perform node actions`.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ClusterNodes/clusterNodes.service.tsx [hunk 1]
```diff
@@ -83,3 +83,7 @@ export const getClusterManifest = (terminalAccessId: number): Promise<ResponseTy
 export const getClusterEvents = (terminalAccessId: number): Promise<ResponseType> => {
     return get(`${Routes.CLUSTER_TERMINAL}/${Routes.POD_EVENTS}?terminalAccessId=${terminalAccessId}`)
 }
+
+export const updateTaints = (taintData: any): Promise<ResponseType> => {
```
Comment (sohel-rp) [inline src/components/ClusterNodes/clusterNodes.service.tsx:87]: Provide request type

File: src/components/ClusterNodes/NodeActions/validationRules.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export class ValidationRules {
+    taintKey = (key: string): { message: string | null; isValid: boolean } => {
+        const keyPrefixRegex = new RegExp(PATTERNS.KUBERNETES_KEY_PREFIX)
+        const keyNameRegex = new RegExp(PATTERNS.KUBERNETES_KEY_NAME)
+
+        if (!key) {
+            return { message: 'Key is required', isValid: false }
+        } else {
+            if (key.length > 253) {
+                return { message: 'Maximum 253 char is allowed', isValid: false }
```
Comment (sohel-rp) [inline src/components/ClusterNodes/NodeActions/validationRules.ts:12]: Shouldn't it be `Maximum 253 chars are allowed`?

File: src/components/ClusterNodes/NodeActions/validationRules.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            } else if (key.indexOf('/') !== -1) {
+                const keyArr = key.split('/')
+                if (keyArr.length > 2) {
+                    return { message: 'Maximum 1 ( / ) allowed', isValid: false }
+                } else if (!keyPrefixRegex.test(keyArr[0])) {
+                    return { message: `Invalid prefix in key`, isValid: false }
```
Comment (sohel-rp) [inline src/components/ClusterNodes/NodeActions/validationRules.ts:18]: use `''`

File: src/components/ClusterNodes/NodeActions/validationRules.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                } else if (!keyNameRegex.test(keyArr[1])) {
+                    return { message: `Invalid name in key`, isValid: false }
```
Comment (sohel-rp) [inline src/components/ClusterNodes/NodeActions/validationRules.ts:20]: same here & below

File: src/components/ClusterNodes/NodeActions/validationRules.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            }
+        }
+        return { message: null, isValid: true }
+    }
+
+    taintValue = (value: string) => {
+        const valueRegex = new RegExp(PATTERNS.KUBERNETES_VALUE)
+        if (value) {
+            if (value.length > 63) {
+                return { message: 'Maximum 63 char is allowed', isValid: false }
```
Comment (sohel-rp) [inline src/components/ClusterNodes/NodeActions/validationRules.ts:33]: same here

File: src/components/ClusterNodes/NodeActions/validationRules.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            } else if (!valueRegex.test(value)) {
+                return { message: `Invalid value`, isValid: false }
```
Comment (sohel-rp) [inline src/components/ClusterNodes/NodeActions/validationRules.ts:35]: same here

File: src/components/ClusterNodes/NodeActions/EditTaintsModal.tsx [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+                                </div>
+                            </div>
+                        )
+                    })}
+                </div>
+                <div className="dc__border-top flex right p-16">
+                    <button className="cta cancel h-36 lh-36 mr-12" type="button" onClick={onClose}>
+                        {EDIT_TAINTS_MODAL_MESSAGING.Actions.cancel}
+                    </button>
+                    <button className="cta h-36 lh-36" onClick={onSave}>
```
Comment (sohel-rp) [inline src/components/ClusterNodes/NodeActions/EditTaintsModal.tsx:223]: disable both buttons when `apiCallInProgress` is true

File: src/components/ClusterNodes/NodeActions/validationRules.ts (hunk 2 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/NodeActions/validationRules.ts:12]: Done

File: src/components/ClusterNodes/NodeActions/validationRules.ts (hunk 3 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/NodeActions/validationRules.ts:18]: Done

File: src/components/ClusterNodes/NodeActions/validationRules.ts (hunk 4 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/NodeActions/validationRules.ts:20]: Done

File: src/components/ClusterNodes/NodeActions/validationRules.ts (hunk 5 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/NodeActions/validationRules.ts:33]: Done

File: src/components/ClusterNodes/NodeActions/validationRules.ts (hunk 6 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/NodeActions/validationRules.ts:35]: Done

File: src/components/ClusterNodes/NodeActions/EditTaintsModal.tsx (hunk 7 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/NodeActions/EditTaintsModal.tsx:223]: Done

File: src/components/ClusterNodes/clusterNodes.service.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/clusterNodes.service.tsx:87]: Done

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**